src/Interpreters/SystemLog.cpp

```diff
@@ -503,6 +503,7 @@ void SystemLog<LogElement>::prepareTable()
             rename->elements.emplace_back(std::move(elem));
 
             auto query_context = Context::createCopy(context);
+            query_context->setSetting("check_referential_table_dependencies", Field{false});
```

Comment:
I suppose this change is ok because this operation is performed automatically and this is a system table. Please write a comment here, and also set `check_table_dependencies` to false for consistency.